src/Type/TypeUtils.php

```diff
@@ -93,6 +93,8 @@ public static function getConstantArrays(Type $type): array
 
 	/**
 	 * @return ConstantStringType[]
+	 *
+	 * @deprecated Use PHPStan\Type\Type::getConstantStrings() instead
```

Comment:
@staabm The two methods don't work exactly the same (i.e. the new implementation for `UnionType` doesn't have `$stopOnUnmatched` equivalent). I think it causes a regression here: https://phpstan.org/r/33df1d6b-50e1-4aed-ac3b-bb5f1a8ac66b (constant string concat with `int|null` - the result shouldn't be a constant string).

Reply:
thx for the headsup. I will have a look